hal/src/dmac/channel/reg.rs

```diff
@@ -256,6 +256,7 @@ reg_proxy!(swtrigctrl, bit, rw);
 /// Acts as a proxy to the PAC DMAC object. Only registers and bits
 /// within registers that should be readable/writable by specific
 /// [`Channel`]s are exposed.
+#[allow(dead_code)]
 pub(super) struct RegisterBlock<Id: ChId> {
```

Comment:
In both the `gpio::v2` and `clock::v2` modules, we seem to have settled on an architecture where a `Registers` struct acts to split a PAC peripheral, like a `GPIO` or an `OSCCTRL`, into individual pieces that are owned, `Send` and `Sync`, like `Pin`s and different oscillator types. On top of that, these structs provide a "task-centric" API rather than a "register-centric" API, if that makes sense.

This implementation strays from that convention. Each `RegisterBlock` *does* represent an individual piece of the full peripheral, but each piece is still `!Sync`, because it stores a `DMAC`. It also has a very register-centric API.

It's not really a big deal, but if you don't have a good reason to keep it this way, I think it might be helpful in the long run to stick with a similar convention across the HAL. Just a thought.

Reply:
I can see the reasoning behind that. I'll probably change it in the future, but I don't think this really falls within the scope of this PR

Reply:
Sure, it was just something that caught my eye.